packages/devtools_app/lib/src/logging/vm_service.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:vm_service/vm_service.dart';
+
+import '../core/message_bus.dart';
+import '../globals.dart';
+
+/// A class which listens for all traffic over the VM service protocol and logs
+/// the traffic to the message bus.
+///
+/// The messages are then available in the Logging page. This class is typically
+/// only used during development by engineers working on DevTools.
+class VmServiceTrafficLogger {
+  VmServiceTrafficLogger(VmService service) {
+    _sendSub = service.onSend.listen(_logServiceProtocolCalls);
+    _receiveSub = service.onReceive.listen(_logServiceProtocolResponses);
+  }
+
+  StreamSubscription _sendSub;
+  StreamSubscription _receiveSub;
+
+  void _logServiceProtocolCalls(String message) {
+    final Map m = jsonDecode(message);
+
+    final String method = m['method'];
+    final String id = m['id'];
+
+    messageBus.addEvent(BusEvent(
+      'devtools.debugger',
```

Comment:
will vm service traffic coming in over logs only ever be for the debugger page? or will we get events for other pages as well (e.g. debugger.timeline or debugger.memory)?